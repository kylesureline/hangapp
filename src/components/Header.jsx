import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { openSidebar } from '../actions/settings';

export const Header = ({ openSidebar }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__content">
        <Link className="header__title" to="/play">
          <h1>Hangapp</h1>
        </Link>
        <div className="trigger-container trigger-container--open show-for-mobile" onClick={openSidebar}>
          <div className="meat" />
          <div className="meat" />
          <div className="meat" />
        </div>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  openSidebar: () => dispatch(openSidebar())
});

export default connect(undefined, mapDispatchToProps)(Header);
