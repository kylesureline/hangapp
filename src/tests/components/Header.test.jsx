import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header.jsx';

let openSidebar, wrapper;

beforeEach(() => {
  openSidebar = jest.fn();
  wrapper = shallow(
    <Header
      openSidebar={openSidebar}
    />
  );
});

test('should render Header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call open sidebar', () => {
  wrapper.find('.trigger-container').prop('onClick')();
  expect(openSidebar).toHaveBeenCalled();
});
