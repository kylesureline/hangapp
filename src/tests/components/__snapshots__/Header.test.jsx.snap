// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        to="/play"
      >
        <h1>
          Hangapp
        </h1>
      </Link>
      <div
        className="trigger-container trigger-container--open show-for-mobile"
        onClick={[MockFunction]}
      >
        <div
          className="meat"
        />
        <div
          className="meat"
        />
        <div
          className="meat"
        />
      </div>
    </div>
  </div>
</header>
`;
